runtime/core/exec_aten/util/scalar_type_util.h

```diff
@@ -998,11 +998,14 @@ struct promote_types {
   ET_INTERNAL_SWITCH_CASE(                                                   \
       ::executorch::aten::ScalarType::ADDITIONAL2, CTYPE_ALIAS, __VA_ARGS__)
 
-#define ET_INTERNAL_SWITCH_CASE_REAL_TYPES_AND3(             \
-    ADDITIONAL1, ADDITIONAL2, ADDITIONAL3, CTYPE_ALIAS, ...) \
-  ET_INTERNAL_SWITCH_CASE_REAL_TYPES_AND2(                   \
-      ADDITIONAL1, ADDITIONAL2, CTYPE_ALIAS, __VA_ARGS__)    \
-  ET_INTERNAL_SWITCH_CASE(                                   \
+#define ET_INTERNAL_SWITCH_CASE_REAL_TYPES_AND3(                             \
+    ADDITIONAL1, ADDITIONAL2, ADDITIONAL3, CTYPE_ALIAS, ...)                 \
+  ET_INTERNAL_SWITCH_CASE_REAL_TYPES(CTYPE_ALIAS, __VA_ARGS__)               \
+  ET_INTERNAL_SWITCH_CASE(                                                   \
+      ::executorch::aten::ScalarType::ADDITIONAL1, CTYPE_ALIAS, __VA_ARGS__) \
+  ET_INTERNAL_SWITCH_CASE(                                                   \
+      ::executorch::aten::ScalarType::ADDITIONAL2, CTYPE_ALIAS, __VA_ARGS__) \
+  ET_INTERNAL_SWITCH_CASE(                                                   \
       ::executorch::aten::ScalarType::ADDITIONAL3, CTYPE_ALIAS, __VA_ARGS__)
```

Comment:
just out of curiosity, why did we need to change this?

Reply:
Good question, I just thought using recursive macros may take longer time to preprocess.
Also, for consistency: `ET_INTERNAL_SWITCH_CASE_REAL_TYPES_AND2 ` doesn't use the `_AND` variant recursively, so why would `_AND3` one do that.